bwaMem.wdl

```diff
@@ -0,0 +1,487 @@
+        memory:  "~{jobMemory} GB"
+        cpu:     "~{threads}"
+        timeout: "~{timeout}"
+    }  
+    
+    output {
+        File outputBam = "~{resultBam}"
+    }
+
+    meta {
+        output_meta: {
+            outputBam: "output bam aligned to genome"
+        }
+    }
+
+}
+
+task bamMerge{
+    input {
+        Array[File] bams
+        String outputFileNamePrefix
+        String bwaMemSuffix = "annotated"
+        Int   jobMemory = 32
+        String modules  = "samtools/1.9"
+        Int timeout     = 72
+    }
+    parameter_meta {
+        bams:  "Input bam files"
+        outputFileNamePrefix: "Prefix for output file"
+        bwaMemSuffix: "Suffix for the output bam file"
+        jobMemory: "Memory allocated indexing job"
+        modules:   "Required environment modules"
+        timeout:   "Hours before task timeout"    
+    }
+
+    String resultMergedBam = "~{outputFileNamePrefix}.~{bwaMemSuffix}.bam"
+
+    command <<<
+        samtools merge \
+        -c -p \
+        ~{resultMergedBam} \
+        ~{sep=" " bams} 
+    >>>
+
+    runtime {
+        memory: "~{jobMemory} GB"
+        modules: "~{modules}"
+        timeout: "~{timeout}"
+    }
+
+    output {
+        File outputMergedBam = "~{resultMergedBam}"
+    }
+
+    meta {
+        output_meta: {
+            outputMergedBam: "output merged bam aligned to genome"
+        }
+    }       
+}
+
+task indexBam {
+    input {
+        File  inputBam
+        Int   jobMemory = 12
+        String modules  = "samtools/1.9"
+        Int timeout     = 48
+    }
+    parameter_meta {
+        inputBam:  "Input bam file"
+        jobMemory: "Memory allocated indexing job"
+        modules:   "Modules for running indexing job"
+        timeout:   "Hours before task timeout"
+    }
+
+    String resultBai = "~{basename(inputBam)}.bai"
+
+    command <<<
+        samtools index ~{inputBam} ~{resultBai}
+    >>>
+
+    runtime {
+        memory: "~{jobMemory} GB"
+        modules: "~{modules}"
+        timeout: "~{timeout}"
+    }
+
+    output {
+        File outputBai = "~{resultBai}"
+    }
+
+    meta {
+        output_meta: {
+            outputBai: "output index file for bam aligned to genome"
+        }
+    }
+
+}
+
+task adapterTrimmingLog {
+    input {
+        Array[File] inputLogs
+        String outputFileNamePrefix
+        Int   numChunk
+        Int   jobMemory = 12
+        Int timeout     = 48
+
+    }
+    parameter_meta {
+        inputLogs:  "Input log files"
+        outputFileNamePrefix: "Prefix for output file"
+        numChunk: "Number of chunks to split fastq file"
+        jobMemory: "Memory allocated indexing job"
+        timeout:   "Hours before task timeout"
+    }
+
+    String allLog = "~{outputFileNamePrefix}.txt"
+    String log = "~{outputFileNamePrefix}.log"
+
+    command <<<
+        set -e
+        set -o pipefail
+        awk 'BEGINFILE {print "###################################\n"}{print}' ~{sep=" " inputLogs} > ~{allLog}
```

Comment:
Umm... this should probably be it's own program. Having something to summarise cutadapt logs would be useful on its own and ideally do it in a way without so much AWK.